test/ProviderAdapter.spec.ts

```diff
@@ -17,16 +17,23 @@ describe('ProviderAdapter', () => {
 
   describe('#request', () => {
     it("should forward to the provider's send method if it's legacy", async () => {
-      const provider = { send: mock } as LegacyProvider
+      const provider = { send: (_args: any, callback: any) => callback(null, result) } as Provider
```

Comment:
I think it'd be cool to change the `any` on callback for `Request.Callback`. We can import Request from `'../src/types'` and also change line `20`